src/fsharp/PrettyNaming.fs

```diff
@@ -109,66 +110,185 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
+        let minOperatorNameLength =
+            opCharTranslateTable
+            |> Array.minBy (snd >> String.length)
+            |> snd
+            |> String.length
+
+        fun opName ->
+            // Has this operator name already been decompiled?
+            match decompiledOperators.TryGetValue opName with
+            | true, op -> op
+            | false, _ ->
+                let opNameLen = opName.Length
+                
+                /// Function which decompiles the mangled operator name back into a string of operator characters.
+                /// Returns None if the name contains text which doesn't correspond to an operator;
+                /// otherwise returns Some containing the original operator. 
+                let rec decompile (sb : System.Text.StringBuilder) idx =
+                    // Have we reached the end of 'opName'?
+                    if idx = opNameLen then
+                        Some (sb.ToString())
+                    else
+                        let choice =
+                            opCharTranslateTable
+                            |> Array.tryFind (fun (_, opCharName) ->
+                                // If this operator character name is longer than the remaining piece of 'opName',
+                                // it's obviously not a match.
+                                let opCharNameLen = opCharName.Length
+                                if opNameLen - idx < opCharNameLen then false
+                                else
+                                    // Does 'opCharName' match the current position in 'opName'?
+                                    System.String.Compare (opName, idx, opCharName, 0, opCharNameLen, System.StringComparison.Ordinal) = 0)
+
+                        match choice with
+                        | None -> None (* giveup *)
+                        | Some (opChar, opCharName) ->
+                            // 'opCharName' matched the current position in 'opName'.
+                            // Append the corresponding operator character to the StringBuilder
+                            // and continue decompiling at the index following this instance of 'opCharName'.
+                            sb.Append opChar |> ignore
+                            decompile sb (idx + opCharName.Length)
+
+                let opNamePrefixLen = opNamePrefix.Length
+                let sb =
+                    /// The maximum number of operator characters that could be contained in the
+                    /// decompiled operator given the length of the mangled custom operator name.
+                    let maxPossibleOpCharCount = (opNameLen - opNamePrefixLen) / minOperatorNameLength
+                    System.Text.StringBuilder (maxPossibleOpCharCount)
+
+                // Start decompiling just after the operator prefix.
+                match decompile sb opNamePrefixLen with
```

Comment:
My thought was to have this simply return `string` directly - instead of returning `Some(sb.ToString())` just return `sb.ToString()`, instead of `None` return `opName`